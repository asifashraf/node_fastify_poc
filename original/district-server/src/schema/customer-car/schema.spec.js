const { fetchGraphQL, testDb } = require('../../lib/test-util');
const { customerCarDetails } = require('../../lib/test-fragments');
const { filter } = require('lodash');
const {
  customers: [{ id: customerId }],
} = require('../../../database/seeds/development');

test('customer can resolve defaultCar', async () => {
  const query = `{
    customer(id: "${customerId}") {
      defaultCar { ${customerCarDetails} }
    }
  }`;
  // openInGraphiql(query);
  const {
    data: {
      customer: { defaultCar },
    },
  } = await fetchGraphQL(query);
  expect(defaultCar.isDefault).toBeTruthy();
  expect(defaultCar).toMatchSnapshot();
});

test('customer can resolve cars', async () => {
  const query = `{
    customer(id: "${customerId}") {
      cars { ${customerCarDetails} }
    }
  }`;
  // openInGraphiql(query);
  const {
    data: {
      customer: { cars },
    },
  } = await fetchGraphQL(query);
  expect(cars.length).toEqual(2);
  expect(cars).toMatchSnapshot();
});

test('customer can save car', async () => {
  const query = `mutation {
                   customerSaveCar(car: {
                    customerId: "${customerId}"
                    name: "Lambo 212"
                    brand: "Lamborghini"
                    color: "Black"
                    plateNumber: "212FLY"
                    isDefault: true
                  }) {
                    error
                    errors
                    customer {
                      cars {
                        id
                        name
                        isDefault
                      }
                     }
                   }
                  }`;

  // openInGraphiql(query);
  const {
    data: {
      customerSaveCar: {
        customer: { cars },
      },
    },
  } = await fetchGraphQL(query);

  // should only get 1 car here
  const defaultCars = filter(cars, 'isDefault', true);
  expect(defaultCars.length).toEqual(1);
  expect(defaultCars[0].name).toEqual('Lambo 212');
  expect(cars).toMatchSnapshot();
});

test('newly saved car is automatically set as default', async () => {
  await testDb
    .handle('customer_cars')
    .where('customer_id', customerId)
    .del();

  const query = `mutation {
                   customerSaveCar(car: {
                    customerId: "${customerId}"
                    name: "Lambo 212"
                    brand: "Lamborghini"
                    color: "Black"
                    plateNumber: "212FLY"
                  }) {
                    error
                    errors
                    customer {
                      cars {
                        id
                        name
                        isDefault
                      }
                     }
                   }
                  }`;

  // openInGraphiql(query);

  const {
    data: {
      error,
      customerSaveCar: {
        customer: { cars },
      },
    },
  } = await fetchGraphQL(query);

  expect(error).toBeUndefined();

  // should only get 1 car here
  const defaultCars = filter(cars, 'isDefault', true);
  expect(defaultCars.length).toEqual(1);
  expect(defaultCars[0].name).toEqual('Lambo 212');

  expect(cars.length).toEqual(1);
  expect(cars).toMatchSnapshot();
});
