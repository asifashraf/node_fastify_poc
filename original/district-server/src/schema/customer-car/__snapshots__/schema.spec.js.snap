// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customer can resolve cars 1`] = `
Array [
  Object {
    "brand": "Chevy Camaro",
    "color": "Black",
    "id": "b55b45ed-5738-4f67-ba8d-9409b76a6c42",
    "isDefault": false,
    "name": "Bitchin Camaro",
    "note": "Dont forget your Motley Cr√ºe T-shirt!",
    "plateNumber": "BITCHNB",
  },
  Object {
    "brand": "Maserati",
    "color": "Silver",
    "id": "cb964122-7214-4a97-8845-d6f107896298",
    "isDefault": true,
    "name": "Bourgie Ride",
    "note": "This is a note, there are many like me. . . ",
    "plateNumber": "DEEZNUTZ",
  },
]
`;

exports[`customer can resolve defaultCar 1`] = `
Object {
  "brand": "Maserati",
  "color": "Silver",
  "id": "cb964122-7214-4a97-8845-d6f107896298",
  "isDefault": true,
  "name": "Bourgie Ride",
  "note": "This is a note, there are many like me. . . ",
  "plateNumber": "DEEZNUTZ",
}
`;

exports[`customer can save car 1`] = `
Array [
  Object {
    "id": "9f03ad20-3fc0-4896-8917-35324e810d58",
    "isDefault": true,
    "name": "Lambo 212",
  },
  Object {
    "id": "b55b45ed-5738-4f67-ba8d-9409b76a6c42",
    "isDefault": false,
    "name": "Bitchin Camaro",
  },
  Object {
    "id": "cb964122-7214-4a97-8845-d6f107896298",
    "isDefault": false,
    "name": "Bourgie Ride",
  },
]
`;

exports[`newly saved car is automatically set as default 1`] = `
Array [
  Object {
    "id": "0f231215-a464-4960-9698-cccd252f6001",
    "isDefault": true,
    "name": "Lambo 212",
  },
]
`;
